lxc/init.go

```diff
@@ -219,16 +219,6 @@ func (c *cmdInit) create(conf *config.Config, args []string) (lxd.InstanceServer
 	}
 	req.Config = configMap
 	req.Devices = devicesMap
-
-	if !c.flagNoProfiles && len(profiles) == 0 {
-		if len(stdinData.Profiles) > 0 {
-			req.Profiles = stdinData.Profiles
-		} else {
-			req.Profiles = nil
-		}
-	} else {
-		req.Profiles = profiles
-	}
```

Comment:
This also should be handled by the server rather than the client.
Effectively replacing our current logic that substitutes a `nil` by the default profile.